### PR TITLE
Inputs rework

### PR DESCRIPTION
**Pull request by Norby09**
Worked on inputs to work properly

### CODE REVIEW

File: src/app/languages.service.ts [hunk 1]
```diff
@@ -60,21 +60,32 @@ export class LanguagesService {
...
       });
       return resourceValue[langObj];
     }
   }
 
-  setResourceValue(resourceKey: string, resourceValue: string, lang: string) {
-
+  setResourceValue(resourceKey: string, resourceValue: string, lang?: string) {
+    if (!lang) {
+      const current_element = this.i18n[0].resources.find(elem => elem.name === resourceKey);
```
Comment (danvatamaniuc) [inline src/app/languages.service.ts]: I get this defaults to the first language set in `i18n` if no language is specified.
Could there be scenarios where `i18n` has no languages, and `setResourceValue` is still called without a language? Just asking.

File: src/app/languages.service.ts [hunk 2]
```diff
@@ -60,21 +60,32 @@ export class LanguagesService {
...
+      if (current_element) {
+        current_element.value = resourceValue;
```
Comment (danvatamaniuc) [inline src/app/languages.service.ts]: May be rewritten as `current_element && (current_element.value = resourceValue);

Also, let's respect the naming convention, and change `current_element` to `currentElement`.

File: src/app/languages.service.ts [hunk 3]
```diff
@@ -60,21 +60,32 @@ export class LanguagesService {
...
+      }
+    }
+  }
+  setResourceName(resourceKey: string, name: string) {
+    if (resourceKey) {
+      const current_element = this.i18n[0].resources.find(elem => elem.name === resourceKey);
```
Comment (danvatamaniuc) [inline src/app/languages.service.ts]: Same comment about naming convention.

File: src/app/link/link.component.ts [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import { Component, OnInit, Input } from '@angular/core';
+import {Component, OnInit, Input, OnChanges} from '@angular/core';
```
Comment (danvatamaniuc) [inline src/app/link/link.component.ts]: Missing spaces.

File: src/app/link/link.component.ts [hunk 5]
```diff
@@ -9,29 +9,31 @@ import { Resource } from 'data/resource';
...
       this.resourceName = this.link.text.toString();
-      this.languageAndResources.push( Language.create({ name  : "en", resources : new Array<Resource>(new Resource({name : this.link.text , value : this.link.text}))}) );
+      this.languageAndResources.push( Language.create({ name  : "en", resources : new Array<Resource>(new Resource({name : this.link.text , value : this.inputValue}))}) );
     } else {
       this.languageAndResources.push( Language.create({ name  : "", resources : new Array<Resource>(new Resource({name : this.resourceName , value : ""}))}) );
     }
   }
-
+  ngOnChanges() {
+    this.inputValue = this.languageService.getResourceValue(this.link.text.toString());
```
Comment (danvatamaniuc) [inline src/app/link/link.component.ts]: Same comment about `toString()` - could it be removed?

File: src/app/title/title.component.ts [hunk 6]
```diff
@@ -27,15 +27,17 @@ export class TitleComponent implements OnInit, OnChanges {
...
-      this.languageAndResources.push( Language.create({ name  : "en", resources : new Array<Resource>(new Resource({name : this.title.text , value : this.inputValue}))}) );
+      this.languageAndResources.push( Language.create({ name  : "en", resources : new Array<Resource>(new Resource({name : this.resourceName , value : this.inputValue}))}) );
+      this.resourceName = this.title.text.toString();
     } else {
       this.defaultLanguages = this.languageService.getLanguages();
       this.languageAndResources.push( Language.create({ name  : " ", resources : new Array<Resource>(new Resource({name : this.resourceName , value : this.inputValue}))}) );
     }
   }
-  ngOnChanges() {
+  ngOnChanges(changes: SimpleChanges) {
```
Comment (danvatamaniuc) [inline src/app/title/title.component.ts]: Doesn't look like `changes` is used anywhere.

File: src/app/description/description.component.html [hunk 7]
```diff
@@ -22,7 +22,7 @@ <h4 class="hr"> Description </h4>
         </div>
       </div>
       <label class="col-lg-6 overwritePadding" > Text :
-        <input type="text" class="form-control" placeholder="Enter resourece" [(ngModel)]="lang.resources[0].value" (blur)="onInputResource(lang.resources[0].value)"/>
+        <input type="text" class="form-control" placeholder="Enter resourece" [(ngModel)]="lang.resources[0].value" (change)="detectChange($event,lang.resources[0].name)" (blur)="onInputResource(lang.resources[0].value)"/>
```
[the remark this answers is not in the document]
Comment (Norby09) [reply src/app/description/description.component.html]: In detect changes i update the resource name and resource value if an update is triggered. The resources array will always have one item so we will not see lang.resources[1].

File: src/app/description/description.component.ts [hunk 8]
```diff
@@ -12,30 +12,32 @@ import { Resource } from 'data/resource';
...
 
-  resources;
-  defaultLanguages;
-  selectedLanguage;
-  resourceValue;
-  showDropdown = true;
-  resourceName = "@description" + Math.floor(Math.random() * 10000 + 1);
-  languageAndResources = [];
-  inputValue = '';
+  public defaultLanguages;
```
[the remark this answers is not in the document]
Comment (Norby09) [reply src/app/description/description.component.ts]: Added the types.

File: src/app/description/description.component.ts [hunk 9]
```diff
@@ -12,30 +12,32 @@ import { Resource } from 'data/resource';
...
+  public resourceValue;
+  public showDropdown = true;
+  public resourceName = "@description" + Math.floor(Math.random() * 10000 + 1);
+  public languageAndResources = [];
+  public inputValue: string;
 
   constructor(private languageService: LanguagesService) {}
 
   ngOnInit() {
     if (this.description.text !== "") {
```
[the remark this answers is not in the document]
Comment (Norby09) [reply src/app/description/description.component.ts]: Modified

File: src/app/description/description.component.ts [hunk 10]
```diff
@@ -12,30 +12,32 @@ import { Resource } from 'data/resource';
...
-      this.resourceName = this.description.text.toString();
       this.languageAndResources.push( Language.create({ name  : "en", resources : new Array<Resource>(new Resource({name : this.description.text , value : this.inputValue}))}) );
+      this.resourceName = this.description.text.toString();
```
[the remark this answers is not in the document]
Comment (Norby09) [reply src/app/description/description.component.ts]: You are right. Deleted toString().

File: src/app/itemList/itemList.component.ts [hunk 11]
```diff
@@ -72,7 +70,7 @@ export class ItemListComponent implements OnInit{
         }
       }
     }
-
+    console.log(this.items);
```
[the remark this answers is not in the document]
Comment (Norby09) [reply src/app/itemList/itemList.component.ts]: Deleted

File: src/app/languages.service.ts (hunk 1 above)
Comment (Norby09) [reply src/app/languages.service.ts]: No. At the beggining of the app we push an empty language and populate it later

File: src/app/description/description.component.ts (hunk 10 above)
[the remark this answers is not in the document]
Comment (Norby09) [reply src/app/description/description.component.ts]: Actually i cant delete it because it generates an error else way.

**Pull request merged**